notifications/notifiers/email.py

```diff
@@ -70,6 +70,10 @@ def send_notification(self, email_notification):
             if not will_send:
                 return
 
+            # if we're trying to send an email, but the preference is never, we should just cancel it
+            if self.notification_settings.is_triggered_never:
+                raise CancelNotificationError()
```

Comment:
Seems like the exceptional cases here (this and `user != self.user`) should be grouped together. For that matter, it looks like both of them could be moved above this context processor, which I think would be preferable as it gets the exception/error cases out of the way before the 'meat' of the function